pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -166,38 +166,70 @@ let
     in stdenv.mkDerivation {
       inherit pname version;
 
-      desktopItem = makeDesktopItem {
+      desktopItem = makeDesktopItem ({
         name = applicationName;
         exec = "${launcherName} %U";
         inherit icon;
         desktopName = "${desktopName}${nameSuffix}${lib.optionalString forceWayland " (Wayland)"}";
-        genericName = "Web Browser";
-        categories = [ "Network" "WebBrowser" ];
-        mimeTypes = [
-          "text/html"
-          "text/xml"
-          "application/xhtml+xml"
-          "application/vnd.mozilla.xul+xml"
-          "x-scheme-handler/http"
-          "x-scheme-handler/https"
-          "x-scheme-handler/ftp"
-        ];
+        startupNotify = true;
         startupWMClass = wmClass;
-        actions = {
-          new-window = {
-            name = "New Window";
-            exec = "${launcherName} --new-window %U";
-          };
-          new-private-window = {
-            name = "New Private Window";
-            exec = "${launcherName} --private-window %U";
-          };
-          profile-manager-window = {
-            name = "Profile Manager";
-            exec = "${launcherName} --ProfileManger";
-          };
-        };
-      };
+        terminal = false;
+      } // (if libName == "thunderbird"
+            then {
+              genericName = "Email Client";
+              comment = "Read and write e-mails or RSS feeds, or manage tasks on calendars.";
+              categories = [
+                "Network" "Calendar" "Chat" "ContactManagement" "Email" "Feed"
```

Comment:
The following are printed when building the package:

```
/nix/store/0v4d62kwq4fqrhrwf470xl6vcwkgy0ry-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value item "Calendar" in key "Categories" in group "Desktop Entry" can be extended with another category among the following categories: Office
/nix/store/0v4d62kwq4fqrhrwf470xl6vcwkgy0ry-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value item "ContactManagement" in key "Categories" in group "Desktop Entry" can be extended with another category among the following categories: Office
```

Reply:
Adding `Office` to `Categories` results in the following message:

```
/nix/store/q5q7kzw09sczj0kjjzcvc7lp3xs9dsnz-thunderbird.desktop/share/applications/thunderbird.desktop: hint: value "Network;Calendar;Chat;ContactManagement;Email;Feed;GTK;News;Office" for key "Categories" in group "Desktop Entry" contains more than one main category; application might appear more than once in the application menu
```

We have three options:

- A. Let Thunderbird appears under both `Network` and `Office`.
- B. Remove `Calendar` and `ContactManagement` from the category.
- C. Keep `Calendar` and `ContactManagement` without `Office`.

Reply:
A. would be weird, I think.